lib/std/os/bits/wasi.zig

```diff
@@ -12,6 +12,8 @@ pub const timespec = extern struct {
     tv_nsec: isize,
 };
 
+pub const PATH_MAX = 4096; // TODO verify this!
```

Comment:
Note that `PATH_MAX` as defined by the zig standard library, is the value for which a file path greater than this length would return `ENAMETOOLONG` from the OS. The OS in this case being of course the WASI interpreter runtime.

Reply:
Yep, thanks for the pointer. So in WASI, we don't have a strict limit on path length AFAIK. The limits are applied by the underlying host instead, so value `4096` seems like a good initial value.

Reply:
This is a bit of a leak of system internal details into WASI which is supposed to provide an operating system agnostic virtual machine. WASI should mandate a maximum length here, otherwise it will be possible to learn facts about the system based on the max path length that is allowed without returning ENAMETOOLONG. Maybe I'm misunderstanding something, please correct me if I'm wrong.

Reply:
Hmm, you're absolutely right. In the refactored code I've got rid of the constant as it's not needed. I used it initially to prototype some of the code while still learning zig.